dev/sg/internal/std/output.go

```diff
@@ -0,0 +1,128 @@
+// Package std defines stdout and related outputting utilities.
+package std
+
+import (
+	"io"
+	"os"
+
+	"github.com/charmbracelet/glamour"
+
+	"github.com/sourcegraph/sourcegraph/lib/output"
+)
+
+// Output is a wrapper with convenience functions for sg output.
+type Output struct {
+	*output.Output
+
+	// Buildkite indicates we are in a Buildkite environment.
+	Buildkite bool
+}
+
+// Out is the standard output which is instantiated when sg gets run.
+var Out *Output
+
+// NewOutput instantiates a new output instance for local use, such as to get
+func NewOutput(dst io.Writer, verbose bool) *Output {
+	return &Output{
+		Output: output.NewOutput(dst, output.OutputOpts{
+			ForceColor: true,
+			ForceTTY:   true,
+			Verbose:    verbose,
+		}),
+		Buildkite: os.Getenv("BUILDKITE") == "true",
+	}
+}
+
+// writeExpanded writes a line that is prefixed Buildkite log output management stuffs such
+// that subsequent lines are expanded.
+//
+// Learn more: https://buildkite.com/docs/pipelines/managing-log-output
+func (o *Output) writeExpanded(line output.FancyLine) {
+	if o.Buildkite {
+		line.Prefix = "+++"
+	}
+	o.WriteLine(line)
+}
+
+// WriteHeading writes a line that is prefixed Buildkite log output management stuffs such
+// that subsequent lines are collapsed if we are in Buildkite.
+//
+// Learn more: https://buildkite.com/docs/pipelines/managing-log-output
+func (o *Output) writeCollapsed(line output.FancyLine) {
```

Comment:
```suggestion
func (o *Output) writeHeading(line output.FancyLine) {
```

Collapsed is `~~~`, headings just happen to be collapsed unless there is an error.

Reply:
I think the naming here is correct, direct from the buildkite docs:

> You can group and collapse your build output by echoing --- [group name] in your build output.

Reply:
`~~~` would be `writeDemphasized`, though I don't think there's a need for it yet:

> de-emphasize your build output by echoing ~~~ [group name] in your build output.

Reply:
I stand corrected! Thanks 🙏